portal-impl/src/com/liferay/portlet/messageboards/service/impl/MBMessageLocalServiceImpl.java

```diff
@@ -1979,11 +1979,13 @@ protected String getDiscussionMessageSubject(String subject, String body)
 
 		subject = HtmlUtil.extractText(body);
 
-		if (subject.length() <= 50) {
-			return subject;
+		int pos = Math.min(subject.length(), 50);
+
+		if (subject.length() > pos) {
+			return subject.substring(0, pos) + StringPool.TRIPLE_PERIOD;
```

Comment:
The interesting change is substring(0, 50) instead of substring(50). While the former gets the first 50 chars (indexed from 0 to 49th position), the latter just returns the complementary string (from char in 50th position to the end).